src/main/java/ladder/LadderMain.java

```diff
@@ -19,13 +19,15 @@ public static void main(String[] args) {
 
         OutputView.printLadderGame(new LadderGameDto(names, ladder, rewards));
 
-        while (true) {
-            String name = InputView.readName();
+        readNameAndPrintResult(results);
+    }
 
-            checkPrintAllAndEnd(name, results);
+    private static void readNameAndPrintResult(Results results) {
+        String name = InputView.readName();
+        checkPrintAllAndEnd(name, results);
+        OutputView.printResult(results.resultOf(name));
 
-            OutputView.printResult(results.resultOf(name));
-        }
+        readNameAndPrintResult(results);
```

Comment:
기저조건이 명확하게 있으니 System.exit(0)을 통해 강제로 로직을 종료하기보다는 기저조건을 명시해서 재귀함수를 작성하면 됩니다 ㅎㅎ 기존의 코드는 wanning code로 아마 메서드는 무한으로 실행되기에 예외로만 종료됩니다. 라는 경고 메세지가 나오고 있었을 꺼에요 ㅎ

```suggestion
private static void readNameAndPrintResult(Results results) {
    String name = InputView.readName();
    if (ALL.equalsIgnoreCase(name)) {
        OutputView.printResults(results);
        return;
    }
    
    checkPrintAllAndEnd(name, results);
    OutputView.printResult(results.resultOf(name));
    
    readNameAndPrintResult(results);
}
```